lib/chezmoi/chezmoi.go

```diff
@@ -95,9 +96,23 @@ func sortedEntryNames(entries map[string]Entry) []string {
 	return entryNames
 }
 
+func sortedScriptNames(entries map[string]*Script) []string {
+	scriptNames := []string{}
+	for scriptName := range entries {
+		scriptNames = append(scriptNames, scriptName)
+	}
+	sort.Strings(scriptNames)
+	return scriptNames
+}
+
 func splitPathList(path string) []string {
 	if strings.HasPrefix(path, string(filepath.Separator)) {
 		path = strings.TrimPrefix(path, string(filepath.Separator))
 	}
 	return strings.Split(path, string(filepath.Separator))
 }
+
+// StripTemplateExtension removes the template extension at the end of the passed filename
+func StripTemplateExtension(name string) string {
+	return strings.TrimSuffix(name, templateSuffix)
+}
```

Comment:
I suspect that we should be have `ScriptAttributes` and `Script` structs in the same way as we have a [`FileAttributes` and `File` structs](https://github.com/twpayne/chezmoi/blob/master/lib/chezmoi/file.go#L14-L34) (and also for [directories](https://github.com/twpayne/chezmoi/blob/master/lib/chezmoi/dir.go#L12-L26)), because scripts are effectively a new top-level concept in `chezmoi` but we can make this change in a follow-up PR.